src/repository/track_repo.rs

```diff
@@ -0,0 +1,165 @@
+use {
+    crate::{
+        crypto::{HashAlgorithm, HashValue},
+        interchange::DataInterchange,
+        metadata::{MetadataPath, MetadataVersion, TargetDescription, TargetPath},
+        repository::{RepositoryProvider, RepositoryStorage},
+        Result,
+    },
+    futures_io::AsyncRead,
+    futures_util::{
+        future::{BoxFuture, FutureExt},
+        io::{AsyncReadExt, Cursor},
+    },
+    parking_lot::Mutex,
+    std::sync::Arc,
+};
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum Track {
+    Store {
+        path: MetadataPath,
+        version: MetadataVersion,
+        metadata: String,
+    },
+    FetchFound {
+        path: MetadataPath,
+        version: MetadataVersion,
+        metadata: String,
+    },
+    FetchErr(MetadataPath, MetadataVersion),
+}
+
+impl Track {
+    pub(crate) fn store<T>(meta_path: &MetadataPath, version: &MetadataVersion, metadata: T) -> Self
+    where
+        T: Into<Vec<u8>>,
+    {
+        Track::Store {
+            path: meta_path.clone(),
+            version: version.clone(),
+            metadata: String::from_utf8(metadata.into()).unwrap(),
+        }
+    }
+
+    pub(crate) fn fetch_found<T>(
+        meta_path: &MetadataPath,
+        version: &MetadataVersion,
+        metadata: T,
+    ) -> Self
+    where
+        T: Into<Vec<u8>>,
+    {
+        Track::FetchFound {
+            path: meta_path.clone(),
+            version: version.clone(),
+            metadata: String::from_utf8(metadata.into()).unwrap(),
+        }
+    }
+}
+
+/// Helper Repository wrapper that tracks all the fetches and stores for testing purposes.
```

Comment:
"... tracks all the <ins>metadata</ins> fetches ..."

Reply:
Fixed, thanks.